File: Solutions/910. Smallest Range II/910. Smallest Range II.java
class Solution {
    public int smallestRangeII(int[] nums, int k) {
        Arrays.sort(nums);
        int res=nums[nums.length-1]-nums[0],right=nums[nums.length-1]-k,left=nums[0]+k;
        for(int i=0;i<nums.length-1;++i){
            int m_ax=Math.max(right,nums[i]+k);
            int m_in=Math.min(left,nums[i+1]-k);
            res=Math.min(res,m_ax-m_in);
        }
        return res;
    }
}

File: Solutions/910. Smallest Range II/910. Smallest Range II.go
func smallestRangeII(nums []int, k int) int {
    sort.Ints(nums)
    n:=len(nums)
    res,right,left:=nums[n-1]-nums[0],nums[n-1]-k,nums[0]+k
    for i:=0;i<n-1;i+=1{
        m_ax:=max(right,nums[i]+k)
        m_in:=min(left,nums[i+1]-k)
        res=min(res,m_ax-m_in)
    }
    return res
}
func max(i,j int)int{
    if(i>j){
        return i
    }
    return j
}
func min(i,j int)int{
    if(i<j){
        return i
    }
    return j
}